src/arch/src/aarch64/cache_info.rs

```diff
@@ -120,92 +167,80 @@ impl CacheType {
     }
 }
 
-impl CacheInfo {
-    fn sysfs_cache(&self, index: u8, file_name: &str) -> Result<String> {
-        readln_special(&PathBuf::from(format!(
-            "{}/index{}/{}",
-            self.cache_dir.as_path().display(),
-            index,
-            file_name
-        )))
-    }
-
-    pub fn populate(&mut self, index: u8) -> Result<()> {
-        let mut err_str = String::new();
-
-        // If the cache level or the type cannot be retrieved we stop the process
-        // of populating the cache levels.
-        match self.sysfs_cache(index, "level") {
-            Ok(level) => {
-                self.level = level
-                    .parse::<u8>()
-                    .map_err(|e| Error::InvalidCacheAttr("level".to_string(), e.to_string()))?;
-            }
-            Err(e) => {
-                // If we cannot read the cache level even for the first level of cache, we will
-                // stop processing anymore cache info and log an error.
-                warn!("Could not read cache level for index {}: {}", index, e);
-                return Err(Error::MissingCacheConfig);
-            }
+fn populate_cache_entry(index: u8, store: &dyn CacheStore) -> Result<CacheEntry> {
```

Comment:
would it make sense to define this method as a constructor for `CacheEntry` ? `from_index()` maybe ?

Reply:
Done